fixups/RegLegacyFixups/InitializeFixup.cpp

```diff
@@ -117,12 +121,102 @@ void LogString(const wchar_t* name, const wchar_t* value)
+
+    std::wregex dependency_version_regex = std::wregex(L"%dependency_version%");
+    std::wregex dependency_path_regex = std::wregex(L"%dependency_root_path%");
+
+    Package dependency = nullptr;
+
+#ifdef _DEBUG
+    Log("Filtering required dependency\n");
+#endif
+    for (auto&& dep : dependencies) {
+        std::wstring_view name = dep.Id().Name();
+#ifdef _DEBUG
+        Log("Checking package dependency %.*LS\n", name.length(), name.data());
+#endif
+
+        if (name.find(dependency_name) != std::wstring::npos) {
+#ifdef _DEBUG
+            Log("Found required dependency\n");
+#endif
+            dependency = dep;
+            break;
+        }
+    }
+    if (dependency == nullptr) 
+    {
+        Log("No package with name %.*LS found\n", static_cast<int>(dependency_name.length()), dependency_name.data());
+        return;
+    }
 
-    Log("Initializing RegLegacyFixups\n");
-    
+    std::wstring dependency_path(dependency.EffectivePath());
+    auto version = dependency.Id().Version();
+    std::wstring dependency_version = std::to_wstring(version.Major) + L"." + std::to_wstring(version.Minor) + L"." + std::to_wstring(version.Build) + L"." + std::to_wstring(version.Revision);
+    Log("Dependency path: %LS, version: %LS\n", dependency_path.c_str(), dependency_version.c_str());
+
+    for (const auto& reg_entry : entries)
+    {
+        HKEY res;
+        std::wstring reg_path = std::regex_replace(reg_entry.path, dependency_version_regex, dependency_version);
+
+        LSTATUS st = ::RegCreateKeyExW(HKEY_CURRENT_USER, reg_path.c_str(), 0, NULL, REG_OPTION_VOLATILE, KEY_ALL_ACCESS, NULL, &res, NULL);
+        if (st != ERROR_SUCCESS)
+        {
+            Log("Create key %LS failed\n", reg_entry.path.c_str());
+            return;
+        }
+
+        g_regRedirectedHivePaths.insert(narrow(reg_path));
+        g_regCreatedKeysOrdered.push_back(reg_path);
+
+        for (const auto& it : reg_entry.values)
+        {
+            std::wstring value_data = std::regex_replace(it.second, dependency_path_regex, dependency_path);
+            value_data = std::regex_replace(value_data, dependency_version_regex, dependency_version);
+
+            st = ::RegSetValueExW(res, it.first.c_str(), 0, REG_SZ, (LPBYTE)value_data.c_str(),
+                (DWORD)(value_data.size() + 1 /* for null termination char */) * sizeof(wchar_t));
+
+            if (st != ERROR_SUCCESS)
+            {
+#ifdef _DEBUG
+                Log("set value %s failed\n", it.first.c_str());
+#endif
+            }
+        }
+    }
+}
+
+void CleanupFixups()
+{
+    Log("RegLegacyFixups CleanupFixups()\n");
+
+    // Deleted the created keys in reverse order
+    std::vector<std::wstring>::reverse_iterator redirected_path = g_regCreatedKeysOrdered.rbegin();
+
+    while (redirected_path != g_regCreatedKeysOrdered.rend())
+    {
+        LSTATUS st = ::RegDeleteTreeW(HKEY_CURRENT_USER, redirected_path->c_str());
+        if (st != ERROR_SUCCESS)
```

Comment:
Please use Windows Implementation Library for testing against error values.

Reply:
This is LSTATUS not HRESULT. Which wil API do you use to check this?